go/vt/vtgate/evalengine/format.go

```diff
@@ -189,7 +189,6 @@ func (b *BitwiseNotExpr) format(buf *formatter, depth int) {
 func (c *ConvertExpr) format(buf *formatter, depth int) {
 	buf.WriteString("CONVERT(")
 	c.Inner.format(buf, depth)
-	buf.WriteString(", ")
```

Comment:
This was causing a duplicate `, ` in the converted output. I'm not sure how this is even working today, but this was generated invalid SQL since the switch below which adds the type also prefixes it with `, `. 

There didn't seem to be any test coverage for this, so I added tests also to verify the newly added binary behavior.